base/dict.jl

```diff
@@ -116,6 +116,14 @@ Dict(ps::Pair...)                  = Dict(ps)
 
 Dict(kv) = dict_with_eltype((K, V) -> Dict{K, V}, kv, eltype(kv))
 
+function getproperty(d::Dict, x::Symbol)
+    if x in propertynames(d)
```

Comment:
I will have to check if incoming symbol `x` is in the propertynames. This is where overhead will be. Other option is to change every occurance of `getproperty` with `getfield` in `dict.jl`. That is more involved and if other packages are trying to check fields of `Dict`, It will be a problem.